### PR TITLE
Reg alloc: error message when a bank is empty

### PR DESCRIPTION
**Pull request by eponier**
Fixes https://github.com/jasmin-lang/jasmin/issues/693

### CODE REVIEW

**Comment by eponier** [comment]
If there is no conflict, this probably means that the bank is empty, so another fix could be to check if the set of conflicts is empty.

File: compiler/tests/fail/register_allocation/arm-m4/no_mmx.jazz [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export fn main () -> reg u32 {
+  reg u32 res;
+  #mmx reg u32 tmp;
```
Comment (vbgl) [inline compiler/tests/fail/register_allocation/arm-m4/no_mmx.jazz:3]: I disagree that this test should fail. The annotation could be ignored, with a warning.

File: compiler/src/regalloc.ml [hunk 2]
```diff
@@ -743,6 +743,17 @@ let two_phase_coloring
   let schedule =
     if !Glob_options.lazy_regalloc then lazy_scheduling variables a
     else schedule_coloring size variables cnf a in
+  (* Give a specific error message if the bank is empty: there is no way the
+     variables can be allocated. We pick one of the variables to illustrate
+     the error message. *)
+  begin match schedule, registers with
```
Comment (vbgl) [inline compiler/src/regalloc.ml:749]: There is no point in computing a schedule. You may pick any `x` in the `variables` hash table (e.g. using `to_seq` and then `uncons`).

File: compiler/tests/fail/register_allocation/arm-m4/no_mmx.jazz (hunk 1 above)
Comment (bgregoir) [reply compiler/tests/fail/register_allocation/arm-m4/no_mmx.jazz:3]: Hi Vincent, I am relatively agree with you. If a mmx annotation does not make sens it is certainly better to fail/ignore it in pretyping instead. Question: why do you think it should be a warning (portability ?)

Comment (vbgl) [reply compiler/tests/fail/register_allocation/arm-m4/no_mmx.jazz:3]: No, it’s the opposite of portability. For portability, you would like any `mmx` annotation to be silently ignored on platforms where it does not make sense. So maybe no warning would be better.

**Review by vbgl — APPROVED** [review]

**Pull request merged**